### ISSUE TITLE
Missing title and artist information in the sidebar

### ISSUE THREAD

**Issue by DavidWiesner** [tiliado/nuvola-app-amazon-cloud-player#5]
Hi,
thanks for your work!
I found an issue with your integration. The method 'update' in 'integrate.js' is unable to fetch information about the title and the artist of the current song. 

![acp-missing title and artist](https://cloud.githubusercontent.com/assets/243115/11029486/9ed8f74c-86c7-11e5-9800-ff5067f37042.png)

I don´t know what webengine nuvola 3 beta use but 'innerText' on a text-element is empty. Similar to the [nuvola-app-google-play/../integrate.js#L183](https://github.com/tiliado/nuvola-app-google-play/blob/master/integrate.js#L183) I recommend using 'textContent' and 'innerText' to determine the text content of an text element ([see also http://perfectionkills.com/the-poor-misunderstood-innerText/](http://perfectionkills.com/the-poor-misunderstood-innerText/)).

Here is an example patch:

```
--- integrate-old.js    2015-11-09 09:37:56.637256248 +0100
+++ integrate.js    2015-11-09 09:47:06.486362774 +0100
@@ -131,8 +131,11 @@

     try {
         var songDetails = playerRoot.getElementsByClassName("currentSongDetails")[0];
-        track.title = songDetails.getElementsByClassName("title")[0].innerText;
-        track.artist = songDetails.getElementsByClassName("artistLink")[0].innerText;
+        var ele;
+        ele = songDetails.getElementsByClassName("title")[0];
+        track.title = ele.textContent || ele.innerText;
+        ele = songDetails.getElementsByClassName("artistLink")[0];
+        track.artist = ele.textContent || ele.innerText;

         var albumImage = playerRoot.getElementsByClassName("albumImage")[0];
         track.artLocation = albumImage.src;
```

regards,
David

---

Amazon Cloud Player: 5.0
Nuvola Player 3 Beta: 3.0.0+201511042023.413b716

**Comment by jiri-janousek** [comment]
Hi @DavidWiesner, thanks for noticing this issue. I guess the HTML code of Amazon Cloud Player changed in a way that made innerText broken. I'll take a look at it later today.

**Comment by DavidWiesner** [comment]
Hi @fenryxo thanks for looking at. But the HTML Code doesn't change, or at least does not effect the parsing. The problem is the webengine and the javascript API to get to the text content of the element. It´s the same element in the DOM, just try the `textContent` and `innerText` to get the inner text

**Comment by jiri-janousek** [comment]
The script used to work fine. Something in the web page must have changed since then and it broke the parsing.

I've made a pull request #6 for @SteVwonder to review.

**Comment by jiri-janousek** [comment]
Fixed in Amazon Cloud Player script 5.1. Thanks for your bug report.
